pkg/check/fileretrieval/fileretrieval.go

```diff
@@ -37,34 +50,65 @@ func Check(c bee.Cluster, o Options) (err error) {
 		for j := 0; j < o.FilesPerNode; j++ {
 			file := bee.NewRandomFile(rnds[i], fmt.Sprintf("%s-%d-%d", o.FileName, i, j), o.FileSize)
 
+			t0 := time.Now()
 			if err := c.Nodes[i].UploadFile(ctx, &file); err != nil {
 				return fmt.Errorf("node %d: %w", i, err)
 			}
+			d0 := time.Since(t0)
+
+			uploadedCounter.WithLabelValues(overlays[i].String()).Inc()
+			uploadTimeGauge.WithLabelValues(overlays[i].String(), file.Address().String()).Set(d0.Seconds())
+			uploadTimeHistogram.Observe(d0.Seconds())
 
 			time.Sleep(1 * time.Second)
+			t1 := time.Now()
 			size, hash, err := c.Nodes[c.Size()-1].DownloadFile(ctx, file.Address())
 			if err != nil {
 				return fmt.Errorf("node %d: %w", c.Size()-1, err)
 			}
+			d1 := time.Since(t1)
+
+			downloadedCounter.WithLabelValues(overlays[i].String()).Inc()
+			downloadTimeGauge.WithLabelValues(overlays[i].String(), file.Address().String()).Set(d1.Seconds())
+			downloadTimeHistogram.Observe(d1.Seconds())
 
 			if !bytes.Equal(file.Hash(), hash) {
+				notRetrievedCounter.WithLabelValues(overlays[i].String()).Inc()
 				fmt.Printf("Node %d. File %d not retrieved successfully. Uploaded size: %d Downloaded size: %d Node: %s File: %s\n", i, j, file.Size(), size, overlays[i].String(), file.Address().String())
 				return errFileRetrieval
 			}
 
+			retrievedCounter.WithLabelValues(overlays[i].String()).Inc()
 			fmt.Printf("Node %d. File %d retrieved successfully. Node: %s File: %s\n", i, j, overlays[i].String(), file.Address().String())
+
+			if metrics {
+				if err := pusher.Push(); err != nil {
+					fmt.Printf("node %d: %s\n", i, err)
+				}
+			}
 		}
 	}
 
 	return
 }
 
 // CheckFull uploads files on cluster and downloads them from the all nodes in the cluster
-func CheckFull(c bee.Cluster, o Options) (err error) {
+func CheckFull(c bee.Cluster, o Options, pusher *push.Pusher, metrics bool) (err error) {
```

Comment:
Is metrics bool redundant? If pusher is not nil, then the metrics are pushed, otherwise not.

Reply:
`pusher` is used in too many places, it was quicker to set metrics